nixos/modules/config/i18n.nix

```diff
@@ -53,8 +53,18 @@ with lib;
 
       supportedLocales = mkOption {
         type = types.listOf types.str;
-        default = [ (config.i18n.defaultLocale + "/UTF-8") ];
-        defaultText = literalExpression "[ (config.i18n.defaultLocale + \"/UTF-8\") ]";
+        default = with config.i18n; lib.unique (
+          (lib.mapAttrsToList (n: v: v + "/UTF-8")
+            (lib.filterAttrs (n: v: n != "LANGUAGE") extraLocaleSettings)
+          ) ++ [ (defaultLocale + "/UTF-8") ]
+        );
```

Comment:
Nit: I think it would be better to build the list of locales first, then append the `/UTF-8` instead of repeating the appending.

Reply:
That would be simpler, yes.